gh-150319: Replace all documentation which says "See PEP 585" (#150325)
python/cpython: Doc/library/stdtypes.rst

   Many other operations also produce lists, including the :func:`sorted`
   built-in.

>    lists are `generic <generics>` over the types of their contents.

Reviewer: ```suggestion
   Lists are `generic <generics>` over the types of their contents.
```

Author: :+1: I just got to that one as I was fixing some typos that `prek` caught. (Also made me run `prek install`, which I was missing!)

> 
>    Lists are :ref:`generic <generics>` over the types of their items.
> 

Reviewer 2: ```suggestion

   Lists are :ref:`generic <generics>` over the types of their items.

```

>    Lists are :ref:`generic <generics>` over the types of their contents.

Reviewer 3: ```suggestion
   Lists are :ref:`generic <generics>` over the types of their items.
```

   Tuples implement all of the :ref:`common <typesseq-common>` sequence
   operations.

   Tuples are :ref:`generic <generics>` over the types of their contents.
   For more information, refer to
   :ref:`the typing documentation on annotating tuples <annotating-tuples>`.

> 

Reviewer 2: ```suggestion
   Tuples are :ref:`generic <generics>` over the types of their contents.
   For more information, refer to
   :ref:`the typing documentation on annotating tuples <annotating-tuples>`.

```

Author: I think this and the comment above are fixing multiline whitespace, but with this one I really can't tell from the github compare view what it's going to do. I'll apply what I _think_ is meant here manually, in a co-authored commit.

Author: I _think_ I got this right in https://github.com/python/cpython/pull/150325/changes/4bb0cbf39846a74485874a7c68bbb0406d40334c ?

But let me know if it's wrong.

Reviewer 2: There are two blank lines after "deques are :ref:`generic <generics>` over the type of their contents" in that commit, so you should remove one. TL;DR: don't add two blank lines most of the time.

Author: I started checking `git diff $(git merge-base HEAD main)` to make sure I wasn't introducing _or removing_ any double-blank-lines from the text.

You'll see that the two blank lines after that section predate these changes.

Should I be removing that? I consider it orthogonal to these changes and would strongly prefer not to be making any change outside of the scope of this work. But I can if it's definitely a bugfix.

Reviewer 2: You can remove blank lines if your new paragraphs fit in bewteen them, but otherwise no need to change elsewhere or add more.

>    Tuples also support the form ``tuple[T, ...]`` to indicate an arbitrary length tuple of elements of type T.

Reviewer 3: I think we should probably just link to our existing in-depth documentation for tuples here (https://docs.python.org/3/library/typing.html#annotating-tuples)

Author: I'd be happy with that, but want to note: this is what's in place for the `__class_getitem__` docstring.

In that light, two questions:
- is the text okay for that docstring? (which you'd only get if you look at something like `help(tuple.__class_getitem__)`)
- should we include this text here in addition to a link to the typing docs?

Author: For the moment, I applied the link as an addition. I'm happy to remove the text, but want to confirm before making that change.

Reviewer 3: The Sphinx docs and the docstring should _usually_ be kept in sync, but in this case I think it would be fine for them to diverge. In the docstring, we can't link to documentation elsewhere in the same way as we can in the Sphinx docs, but in the Sphinx docs it doesn't really make sense to repeat information that's laid out more fully elsewhere, I don't think

Author: Okay, that works for me! Like you say, `__doc__` can't cross-reference things.

In the sphinx doc, I'll trim it down to
```
Tuples are :ref:`generic <generics>` over the types of their contents.
For more information, refer to
:ref:`the typing documentation on annotating tuples <annotating-tuples>`.
```

   Tuples implement all of the :ref:`common <typesseq-common>` sequence
   operations.

   Tuples are `generic <generics>` over the types of their contents.

>    The type arguments to a tuple should match the types and arity of the contents.

Reviewer: I'm not sure this sentence adds much, could we just delete it? (Here and in the corresponding docstring.)

Author: Yeah, with the examples maybe it's unnecessary. Removing.

   types such as :class:`bytes` and :class:`bytearray`, an element is a single
   byte, but other types such as :class:`array.array` may have bigger elements.

>    :class:`!memoryviews` are :ref:`generic <generics>` over the type of their

Reviewer 2: ```suggestion
   :class:`!memoryview`\s are :ref:`generic <generics>` over the type of their
```

      >>> hash(v[::-2]) == hash(b'abcefg'[::-2])
      True

   memoryviews are :ref:`generic <generics>` over the type of their contents, as

>    retrieved by indexing.

Reviewer 3: I'm not sure I'd refer to indexing here, since not all memoryviews are indexable (only if `~memoryview.format` is one of the native format specifiers from the `struct` module).

```suggestion
   memoryviews are :ref:`generic <generics>` over the type of their underlying
   data.
```

Author: I hadn't realized! I'm pretty unfamiliar with these types, and use them very rarely.

Reviewer 3: Heh, I've never used memoryview, I only read the docs just now to review your PR 😆

      >>> hash(v[::-2]) == hash(b'abcefg'[::-2])
      True

>    memoryviews are :ref:`generic <generics>` over the type of their underlying data.

Reviewer 2: I would move this just after

> For many simple
   types such as :class:`bytes` and :class:`bytearray`, an element is a single
   byte, but other types such as :class:`array.array` may have bigger elements.

because here we're already explaining the methods.

Author: :+1: I did exactly this move. I was not sure where to place it -- I've put the generic note uniformly at the end, but this one reads very differently.

:meth:`~set.discard` methods may be a set.  To support searching for an equivalent
frozenset, a temporary one is created from *elem*.

> Sets and frozensets are :ref:`generic <generics>` over the type of their contents.

Reviewer 3: ```suggestion
Sets and frozensets are :ref:`generic <generics>` over the type of their elements.
```

      Dictionary order is guaranteed to be insertion order.  This behavior was
      an implementation detail of CPython from 3.6.

   Dictionaries are :ref:`generic <generics>` over two types, for the types of

>    keys and values, respectively.

Reviewer 3: ```suggestion
   Dictionaries are :ref:`generic <generics>` over two types, signifying
   (respectively) the types of the dictionary's keys and values.
```

   :class:`!frozendict` is not a :class:`!dict` subclass but inherits directly
   from ``object``.

   Like dictionaries, frozendicts are :ref:`generic <generics>` over two types,

>    for the types of keys and values, respectively.

Reviewer 3: ```suggestion
   Like dictionaries, frozendicts are :ref:`generic <generics>` over two types,
   signifying (respectively) the types of the frozendict's keys and values.
```
